### ISSUE TITLE
Dict as type annotation is refused

### ISSUE THREAD

**Issue by tamuhey** [microsoft/pyright#1167]
```python
types = {"foo": str, "bar": int}
a: types["foo"] = "x"
```

raises:

```
2:4 - error: Expected class type but received "Dict[str, Unknown]" (reportGeneralTypeIssues)
```

but the below doesn't raises:

```python
types = {"foo": str, "bar": int}
cls = types["foo"] # assign into a variable before using it as the annotation
a: cls = "x"
```

pyright version: 1.1.85

**Comment by tamuhey** [comment]
This doesn't raises as well:

```python
types = {"foo": str, "bar": int}
a: (_ := types["foo"]) = "x"
```

**Comment by tamuhey** [comment]
According to the [full grammar specification](https://docs.python.org/3/reference/grammar.html), I think   dict should be accepted as an annotation.

**Comment by erictraut** [comment]
Type annotations need to be static types (classes), not objects (class instances). Keep in mind that a static type checker does not "execute" the code like the Python interpreter does. It works on static type declarations. So you would need to use the following.

```python
a: str = "x"
```
